src/Public/OAComponents.ps1

```diff
@@ -0,0 +1,880 @@
+        New-PodeOAServerEndpoint -Url 'http://ext.server.com/api/v12' -Description 'ext test server' |
+        New-PodeOAServerEndpoint -Url 'http://ext13.server.com/api/v12' -Description 'ext test server 13'
+        ) |
+        Set-PodeOARouteInfo -Summary 'Find pets by ID' -Description 'Returns pets based on ID'  -OperationId 'getPetsById' -PassThru |
+        Set-PodeOARequest -PassThru -Parameters @(
+        (New-PodeOAStringProperty -Name 'id' -Description 'ID of pet to use' -array | ConvertTo-PodeOAParameter -In Path -Style Simple -Required )) |
+        Add-PodeOAResponse -StatusCode 200 -Description 'pet response'   -Content (@{ '*/*' = New-PodeOASchemaProperty   -ComponentSchema 'Pet' -array }) -PassThru |
+        Add-PodeOAResponse -Default  -Description 'error payload' -Content (@{'text/html' = 'ErrorModel' }) -PassThru
+#>
+function Add-PodeOAComponentPathItem {
+    param(
+
+        [Parameter(Mandatory = $true)]
+        [ValidatePattern('^[a-zA-Z0-9\.\-_]+$')]
+        [string]
+        $Name,
+
+        [Parameter(Mandatory = $true )]
+        [ValidateSet('Connect', 'Delete', 'Get', 'Head', 'Merge', 'Options', 'Patch', 'Post', 'Put', 'Trace', '*')]
+        [string]
+        $Method,
+
+        [switch]
+        $PassThru,
+
+        [string[]]
+        $DefinitionTag
+    )
+
+    $DefinitionTag = Test-PodeOADefinitionTag -Tag $DefinitionTag
+
+    $refRoute = @{
+        Method      = $Method.ToLower()
+        NotPrepared = $true
+        OpenApi     = @{
+            Responses      = $null
+            Parameters     = $null
+            RequestBody    = $null
+            callbacks      = [ordered]@{}
+            Authentication = @()
+        }
+    }
+    foreach ($tag in $DefinitionTag) {
+        if (Test-OpenAPIVersion -Version 3.0 -DefinitionTag $tag  ) {
+            throw 'The feature reusable component pathItems is not available in OpenAPI v3.0.x'
+        }
+        #add the default OpenApi responses
+        if ( $PodeContext.Server.OpenAPI[$tag].hiddenComponents.defaultResponses) {
+            $refRoute.OpenApi.Responses = $PodeContext.Server.OpenAPI[$tag].hiddenComponents.defaultResponses.Clone()
+        }
+        $PodeContext.Server.OpenAPI[$tag].components.pathItems[$Name] = $refRoute
+    }
+
+    if ($PassThru) {
+        return $refRoute
+    }
+}
+
+
+
+
+
+<#
+.SYNOPSIS
+Check the OpenAPI version
+
+.DESCRIPTION
+Check the OpenAPI version for a specific OpenAPI Definition
+
+
+.PARAMETER Version
+The version number to compare
+
+.PARAMETER DefinitionTag
+An Array of strings representing the unique tag for the API specification.
+This tag helps in distinguishing between different versions or types of API specifications within the application.
+You can use this tag to reference the specific API documentation, schema, or version that your function interacts with.
+
+.EXAMPLE
+Test-OpenAPIVersion -Version 3.1 -DefinitionTag 'default'
+#>
+
+function Test-OpenAPIVersion {
```

Comment:
Should ideally be called `Test-PodeOAVersion`, so it has the "Pode" tag and the "OA" abbreviation.

Reply:
I don't know why I used so weird name.
probably I had the fever and later I forgot to check the name